flake.nix

```diff
@@ -0,0 +1,70 @@
+{
+  description = "rquickshare flake";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
+    # as upstream nixpkgs doesn't have tauri 2, we use a fork
+    nixpkgs2.url = "github:hannesGitH/nixpkgs?ref=tauri2";
+    corelib.url = "./core_lib";
+  };
+
+  outputs = inputs@{ nixpkgs, ... }:
+    let
+      inherit (nixpkgs) lib;
+      # definitely not tested, but i like it exposed, who knows
+      systems = lib.systems.flakeExposed;
+      forAllSystems = lib.genAttrs systems;
+      spkgs = system : nixpkgs.legacyPackages.${system}.pkgs;
+    in
+    {
+      packages = forAllSystems (system: with spkgs system; rec {
+        remote2 = inputs.nixpkgs2.legacyPackages.${system};
+        tauri = remote2.pkgs.cargo-tauri;
+        corelib_pkgs = inputs.corelib.outputs.packages.${system};
+        corelib_src = corelib_pkgs.rqscore_src;
+        corelib_path = "rqs_lib";
+        rquickshare = rustPlatform.buildRustPackage rec {
+          name = "rquickshare";
+          src = ./app/main;
+          nativeBuildInputs = [ tauri.hook pnpm.configHook nodejs moreutils protobuf ];
+          cargoRoot = "src-tauri";
+          cargoLock = {
+            lockFile = app/main/src-tauri/Cargo.lock;
+            outputHashes = {
+              "mdns-sd-0.10.4" = "sha256-y8pHtG7JCJvmWCDlWuJWJDbCGOheD4PN+WmOxnakbE4=";
+            };
+          };
+          pnpmDeps = pnpm.fetchDeps {
+            inherit src;
+            pname = name;
+            hash = "sha256-ko0S934TnGDVbTrFkunHPt0e/pT5CVi4yorTi0WYqmc=";
```

Comment:
And this hash?

Reply:
If your pnpm.lock changes

Similarly to the cargo one, it ensures all dependency are exactly the same. In nix flakes normally accessing the Internet is not allowed, except if you now apriori the resulting hash, it's called a fixed output derivation.